lib/render/SliceRenderer.cpp

```diff
@@ -0,0 +1,700 @@
+);
+
+SliceRenderer::SliceRenderer(
+    const ParamsMgr* pm,
+    string winName,
+    string dataSetName,
+    string instanceName,
+    DataMgr* dataMgr
+) : Renderer(
+    pm, 
+    winName, 
+    dataSetName, 
+    SliceParams::GetClassType(),
+    SliceRenderer::GetClassType(), 
+    instanceName, 
+    dataMgr
+) {
+    _initialized   = false;
+    _textureWidth  = 200;
+    _textureHeight = 200;
+
+    _vertexCoords = {
+        0.0f, 0.0f, 0.f,
+        1.0f, 0.0f, 0.f,
+        0.0f, 1.0f, 0.f,
+        1.0f, 0.0f, 0.f,
+        1.0f, 1.0f, 0.f,
+        0.0f, 1.0f, 0.f
+    };
+
+    _texCoords = {
+        0.0f, 0.0f,
+        1.0f, 0.0f,
+        0.0f, 1.0f,
+        1.0f, 0.0f,
+        1.0f, 1.0f,
+        0.0f, 1.0f
+    };
+
+    _VAO                   = 0;
+    _vertexVBO             = 0;
+    _texCoordVBO           = 0;
+    _colorMapTextureID     = 0;
+    _dataValueTextureID    = 0;
+
+    _cacheParams.domainMin.resize(3, 0.f);
+    _cacheParams.domainMax.resize(3, 1.f);
+
+    SliceParams* p = dynamic_cast<SliceParams*>(GetActiveParams());
+    assert(p);
+    MapperFunction* tf = p->GetMapperFunc(_cacheParams.varName);
+    _colorMapSize = tf->getNumEntries();
+}
+
+SliceRenderer::~SliceRenderer() {
+    if (_VAO != 0) {
+        glDeleteVertexArrays(1, &_VAO);
+        _VAO = 0;
+    }
+
+    if (_vertexVBO != 0) {
+        glDeleteBuffers(1, &_vertexVBO);
+        _vertexVBO = 0;
+    }
+
+    if (_texCoordVBO != 0) {
+        glDeleteBuffers(1, &_texCoordVBO);
+        _texCoordVBO = 0;
+    }
+   
+    if (_colorMapTextureID != 0) { 
+        glDeleteTextures(1, &_colorMapTextureID);
+        _colorMapTextureID = 0;
+    }
+
+    if (_dataValueTextureID != 0) {
+        glDeleteTextures(1, &_dataValueTextureID);
+        _dataValueTextureID = 0;
+    }
+}
+
+int SliceRenderer::_initializeGL() {
+    _initVAO();
+    _initTexCoordVBO();
+    _initVertexVBO();
+
+    _initialized = true;
+
+    return 0;
+}
+
+void SliceRenderer::_initVAO() {
+    glGenVertexArrays(1, &_VAO);
+    glBindVertexArray(_VAO);
+}
+
+void SliceRenderer::_initTexCoordVBO() {
+    if (_texCoordVBO != 0)
+        glDeleteBuffers(1, &_texCoordVBO);
+    
+    glGenBuffers(1, &_texCoordVBO);
+    glBindBuffer(GL_ARRAY_BUFFER, _texCoordVBO);
+    glVertexAttribPointer(1, 2, GL_FLOAT, GL_FALSE, 0, (void*)0);
+    glEnableVertexAttribArray(1);
+    glBufferData(
+        GL_ARRAY_BUFFER,
+        sizeof(float)*sizeof(_texCoords),
+        _texCoords.data(),
+        GL_STATIC_DRAW
+    );
+}
+
+void SliceRenderer::_initVertexVBO() {
+    if (_vertexVBO != 0) 
+        glDeleteBuffers(1, &_vertexVBO);
+    
+    glGenBuffers(1, &_vertexVBO);
+    glBindBuffer(GL_ARRAY_BUFFER, _vertexVBO);
+    glVertexAttribPointer(0, 3, GL_DOUBLE, GL_FALSE, 0, (void*)0);
+    glEnableVertexAttribArray(0);
+    glBufferData(
+        GL_ARRAY_BUFFER, 
+        6*3*sizeof(double), 
+        _vertexCoords.data(), 
```

Comment:
I think the use of a std::vector here is excessive but its up to you if you want to change it